saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
+
+        self.fissile_tank_db = self.f.create_dataset('fissile tank composition', shape, maxshape=maxshape,
+                                                     chunks=True)
+        self.waste_tank_db = self.f.create_dataset('waste tank composition',
                                                    shape, maxshape=maxshape,
                                                    chunks=True)
-        self.noble_adens_db = self.f.create_dataset('noble adensity',
-                                                    shape, maxshape=maxshape,
-                                                    chunks=True)
-        self.th_adens_db = self.f.create_dataset('Th tank adensity',
-                                                 shape, maxshape=maxshape,
-                                                 chunks=True)
-        # !! raffinate stream consider splitting by what element
-        self.rem_adens = np.zeros((5, self.number_of_isotopes))
+        # initial composition
+
         dt = h5py.special_dtype(vlen=str)
-        self.isolib_db = self.f.create_dataset('iso codes', data=self.isolib,
+        # have to encode to utf8 for hdf5 string
+        self.isolib_db = self.f.create_dataset('iso names',
+                                               data=[x.encode('utf8')
+                                                     for x in self.isoname],
                                                dtype=dt)
-
-        self.bu_adens_db_0[0, :] = self.dict_to_array(self.bumat_dict)
-        self.bu_adens_db_1[0, :] = self.dict_to_array(self.bumat_dict)
-
-        self.th232_adens_0 = self.bumat_dict['Th232']
-
-    def dict_to_array(self, bumat_dict):
-        """ Converts an OrderedDict to an array of its values
-
-        Parameters:
-        -----------
-        bumat_dict: OrderedDict
-            key: isotope name
-            value: adensity
-
-        Returns:
-        --------
-        array of bumat_dict values
+        self.isozai_db = self.f.create_dataset('iso zai', data=self.isozai)
+        # the first depleted, non-reprocessed fuel is stored in timestep 1
+        # initial composition
+        self.dep_dict = self.read_dep()
+        self.driver_before_db[0,
+                              :] = self.dep_dict[self.driver_mat_name] * self.driver_vol
+        self.driver_after_db[0,
+                             :] = self.dep_dict[self.driver_mat_name] * self.driver_vol
+        try:
+            self.blanket_before_db[0,
+                                   :] = self.dep_dict[self.blanket_mat_name] * self.blanket_vol
+            self.blanket_after_db[0,
+                                  :] = self.dep_dict[self.blanket_mat_name] * self.blanket_vol
+        except:
+            self.blanket_before_db[0, :] = np.zeros(self.number_of_isotopes)
+            self.blanket_after_db[0, :] = np.zeros(self.number_of_isotopes)
+            print('Blanket not defined: going to be all zeros')
+
+    def write_run_info(self):
```

Comment:
This function doesn't do what's on the tin. It does way more. Break this function up into its functional parts (read the serpent file, pull out important parameters, write certain things to the db.) This will mean your functions will need to return actual data and will need to feed that data to one another.